flink-runtime/src/test/java/org/apache/flink/runtime/dispatcher/DefaultJobManagerRunnerRegistryTest.java

```diff
@@ -200,14 +197,36 @@ public void testFailingLocalCleanupAsync() throws Exception {
                 .failsWithin(Duration.ZERO)
                 .withThrowableOfType(ExecutionException.class)
                 .extracting(FlinkAssertions::chainOfCauses, FlinkAssertions.STREAM_THROWABLE)
-                .hasExactlyElementsOfTypes(
-                        ExecutionException.class,
-                        FlinkException.class,
-                        expectedException.getClass())
+                .hasExactlyElementsOfTypes(ExecutionException.class, expectedException.getClass())
                 .last()
                 .isEqualTo(expectedException);
     }
 
+    @Test
+    public void testLocalCleanupAsyncNonBlocking() {
+        final TestingJobManagerRunner jobManagerRunner =
+                TestingJobManagerRunner.newBuilder().setBlockingTermination(true).build();
+        testInstance.register(jobManagerRunner);
+
+        // this call would block
+        final CompletableFuture<Void> cleanupFuture =
+                Assertions.assertTimeoutPreemptively(
+                        Duration.ofMinutes(10),
+                        () ->
+                                testInstance.localCleanupAsync(
+                                        jobManagerRunner.getJobID(),
+                                        java.util.concurrent.Executors.newSingleThreadExecutor()),
```

Comment:
hmm, I'd rather use some executor here that does not consume resources (because this thread will pollute thread dumps on CI). I thought about a DirectExecutor (its unused anyway so :shrug:), but maybe it should a some kind of FailingExecutor that rejects everything.

Reply:
Yeah, true... it's not getting used. I had some other test implementation in mind when adding the `SingleThreadExecutor`. :facepalm: Good catch :+1: